1 - DataTypes/Code/WeirdCalculator.cs

```diff
@@ -40,28 +42,68 @@ public static decimal ComplexDecimalAdd(float firstValue, double secondValue, de
 			 * If you are new to exceptions, google "C# Exception Handling" - litterally half our jobs is googling :P
 			 */
 
-			throw new NotImplementedException();
+            decimal first;
+            decimal second;
+            
+            try
+            {
+                first = (decimal)firstValue;
+            }
+            catch
+            {
+                return 0;
+            }
+            try
+            {
+                second = (decimal)secondValue;
+            }
+            catch
+            {
+                return 0;
+            }
+
+            return first + second + thirdValue;
```

Comment:
Perfect!

You could have also combined the two catches if you wanted to as well as if either fail, they will return zero. For example:

```
decimal actual = 0;

            try
            {
                actual += (decimal)firstValue;
				actual += (decimal)secondValue;
				actual += thirdValue;
				
				return actual;
            }
            catch
            {
                return 0;
            }
```